app/main/views/audits.py

```diff
@@ -180,6 +212,48 @@ def acknowledge_audit(audit_id):
     return jsonify(auditEvents=audit_event.serialize()), 200
 
 
+@main.route('/audit-events/<int:audit_id>/acknowledge-including-previous', methods=['POST'])
```

Comment:
I think the implementation makes sense, but I'm not sure about the endpoint itself.

We've used "verb" endpoints before, but they're usually tied to a specific object they're operating on (eg `/briefs/<brief_id>/copy`, `/agreeements/<agreement_id>/approve`). This one doesn't though, it affects a collection of objects at once and I don't think a single audit event is a meaningful entry point for this.

First of all, it creates an implicit relationship that hasn't been hardcoded before. The fact that only events with the same type and target object are acknowledged is never expressed by the caller, unlike eg when filtering the events. While that relationship is arguably there, the grouping doesn't necessarily behave the same way for different audit types. Eg I don't think we'd want to acknowledge clarification questions, password resets or failed login events in bulk. But similar to the include earliest flag, an audit-event endpoint defines that behaviour on all event types at once.

From a more practical perspective, by dropping the resource semantics of REST we risk losing some of the benefits it provides. In this case for example, I can imagine it being quite hard to find a log message of the API call that marked a certain audit event as acknowledged. In order to do that we'd need to search the logs for IDs of all subsequent events or find the latest event that was acknowledged at the same time.

These are largely the reasons I tried to push for a separate resource for the service updates. But since we're not doing that one alternative I can see is moving this to something like `POST /services/<id>/updates/acknowledge` with a payload of `{"latest_audit_id": ...}`. This is still a bit awkward, but it immediately limits the audit type to service updates, makes the relationship between affected audit events more explicit and allows us to investigate the logs by only looking up the audit event we're interested in (since it will contain the acknowledged timestamp and the target object id).